src/libraries/Microsoft.Extensions.Options/gen/Parser.cs

```diff
@@ -467,7 +470,81 @@ private List<ValidatedMember> GetMembersToValidate(ITypeSymbol modelType, bool s
                         continue;
                     }
 
-                    var validationAttr = new ValidationAttributeInfo(attributeType.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat));
+                    string attributeFullQualifiedName = attributeType.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat);
+                    if (SymbolEqualityComparer.Default.Equals(attributeType, _symbolHolder.MaxLengthAttributeSymbol))
+                    {
+                        if (!CacheAttributesData(memberType, _symbolHolder.MaxLengthAttributeSymbol.Name, location, attributeType, ref attributeFullQualifiedName))
+                        {
+                            continue;
+                        }
+                    }
+                    else if (SymbolEqualityComparer.Default.Equals(attributeType, _symbolHolder.MinLengthAttributeSymbol))
+                    {
+                        if (!CacheAttributesData(memberType, _symbolHolder.MinLengthAttributeSymbol.Name, location, attributeType, ref attributeFullQualifiedName))
+                        {
+                            continue;
+                        }
+                    }
+                    else if (_symbolHolder.MinLengthAttributeSymbol is not null &&
+                            SymbolEqualityComparer.Default.Equals(attributeType, _symbolHolder.LengthAttributeSymbol))
+                    {
+                        if (!CacheAttributesData(memberType, _symbolHolder.LengthAttributeSymbol.Name, location, attributeType, ref attributeFullQualifiedName))
+                        {
+                            continue;
+                        }
+                    }
+                    else if (SymbolEqualityComparer.Default.Equals(attributeType, _symbolHolder.CompareAttributeSymbol))
+                    {
+                        ImmutableArray<IParameterSymbol> constructorParameters = attribute.AttributeConstructor?.Parameters ?? ImmutableArray<IParameterSymbol>.Empty;
+                        if (constructorParameters.Length == 1 && constructorParameters[0].Name == "otherProperty" && constructorParameters[0].Type.SpecialType == SpecialType.System_String)
+                        {
+                            _commonUtilities.EnsureTrackingAttribute(_symbolHolder.CompareAttributeSymbol.Name, createValue: true, out HashSet<object>? trackedTypeList);
+                            trackedTypeList!.Add((modelType.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat), (string)attribute.ConstructorArguments[0].Value!));
+                            attributeFullQualifiedName = $"{Emitter.StaticGeneratedValidationAttributesClassesNamespace}.{Emitter.StaticAttributeClassNamePrefix}{_commonUtilities.Suffix}_{_symbolHolder.CompareAttributeSymbol.Name}";
+                        }
```

Comment:
Suggestion: Refactor this out to a `TrackCompareAttributeIfNeeded` (or similar) helper method that contains the logic for tracking the attribute if needed.